chapters/intro-computer-architecture/binary-hex/reading/README.md

```diff
@@ -0,0 +1,142 @@
+
+For representing information (instructions and data), computers use the binary system (base 2).
+When writing programs in assembly language, the hexadecimal system (base 16) is preferred because it saves the programmer from writing long strings of 1s and 0s, and conversion to/from binary can be done much more easily than with the decimal system (base 10).
+
+>**NOTE**: We'll use the prefix `0b` for representing numbers in binary and `0x` for hexadecimal numbers.
+> For example, we can write the unsigned integer `127` as `0b01111111` or `0x7F`.
+
+## Binary System
+
+In the **binary system** (base 2), values are represented as a string of 0s and 1s.
+Each digit in the string represents a bit, and a group of 8 bits forms a byte.
+A group of 4 bits is called a **nibble** or **half-byte**.
+
+### Operations with Values Represented in Binary
+
+#### Arithmetic Operations
+
+[Guide for Arithmetic Operations](https://www.tutorialspoint.com/computer_logical_organization/binary_arithmetic.htm)
+
+### Logical Operations
+
+#### Operators on Binary Values
+
+1. **NOT Operation**:
+Inverts each bit.
+
+Example:
+```text
+INV(0b10011010) = 0b01100101
+```
+
+2. **Logical AND Operation**:
+Performs the 'and' operation between bits at the same positions in operands.
+
+Example:
+```text
+0b1001 AND 0b0111 = 0b0001
+```
+
+3. **Logical OR Operation**:
+Performs the 'or' operation between bits at the same positions in operands.
+
+Example:
+```text
+0b1001 OR 0b0111 = 0b1111
+```
+
+4. **Exclusive OR (XOR) Operation**:
+If bits at the same positions in operands have equal values, the resulting bit is 0;
+otherwise, it's 1.
+
+Example:
+```text
+0b1001 XOR 0b0111 = 0b1110
+```
+
+#### Logical Shifts
+
+Logical shifts left/right involve moving each bit by one position.
+Since the result must be on the same number of bits as the initial value, the first bit is lost, and the empty space is filled with a 0 bit.
+
+![Left Logical Shift](../media//left-logical-shift.svg)
+
+![Right Logical Shift](../media//right-logical-shift.svg)
+
+For explanations related to bitwise operations in C, refer to the guide at [Bitwise Operators in C](https://www.geeksforgeeks.org/bitwise-operators-in-c-cpp/).
+
+## Hexadecimal System
+
+In the **hexadecimal system** (base 16), values are represented as a string of characters from '0' to '9' or 'a' to 'f'.
+A byte consists of two such characters, so each character corresponds to a group of 4 bits (a nibble).
+
+### Conversion from Decimal to Binary/Hexadecimal
+
+- Divide the number successively by the base number (2 or 16) and keep the remainders.
+- When the quotient of the division becomes 0, write down the remainders in reverse order.
+- In the case of base 16, when the remainder is greater than 9, letters a-f are used (`0xa = 10`, `0xf = 15`).
+
+#### Example: Conversion of the number 0xD9B1 to decimal
+
+$$ \texttt{0xD9B1} = 1 \cdot 16 ^ 0 + 11 \cdot 16 ^ 1 + 9 \cdot 16 ^ 2 + 13 \cdot 16 ^ 3 = 55729 $$
```

Comment:
This doesn't render for me on the github pages [1]

[1]: https://cs-pub-ro.github.io/hardware-software-interface/1/Lab%201%20-%20Number%20Representation/Binary%20and%20Hexadecimal/Reading/#example-conversion-of-the-number-0xd9b1-to-decimal

Reply:
It should be fixed by adding `math: true` to `config.yaml`, but somehow it isn't. I'll debug this further.